### PR TITLE
Updated to arriba 1.2.0. Added draft conda environment file.

### PR DESCRIPTION
**Pull request by vinjana**
+ Updated readme.

### CODE REVIEW

File: resources/analysisTools/rnaseqworkflow/HIPO2_rnaseq_processing.sh [hunk 1]
```diff
@@ -262,11 +262,20 @@ if [ "$RUN_ARRIBA" == true ]
...
-	if [[ -f "${SAMPLE}_${pid}.fusions.txt" ]]
+
+	arribaCommand="$ARRIBA_BINARY -c '$ALIGNMENT_DIR/$STAR_CHIMERA_MKDUP_BAM' -x '$ALIGNMENT_DIR/$STAR_SORTED_MKDUP_BAM' -a '$GENOME_FA' -k '$ARRIBA_KNOWN_FUSIONS' -g '$GENE_MODELS' -b '$ARRIBA_BLACKLIST' -T -P -o '${SAMPLE}_$pid.fusions.txt' -O '${SAMPLE}_$pid.discarded_fusions.txt'"
+
+    if [ "$useSingleEndProcessing" == true ]; then
+        arribaCommand="$arribaCommand -s yes"
+    else
+        arribaCommand="$arribaCommand -s no"
+    fi
+
```
Comment (suhrig) [inline resources/analysisTools/rnaseqworkflow/HIPO2_rnaseq_processing.sh]: This is superfluous with Arriba 1.2.0. SE/PE is detected automatically. Even in version 0.8 this was not the correct way to specify SE. The option `-s` determines the strandedness (which is also detected automatically, BTW, so no need to specify).

File: resources/analysisTools/rnaseqworkflow/HIPO2_rnaseq_processing.sh [hunk 2]
```diff
@@ -262,11 +262,20 @@ if [ "$RUN_ARRIBA" == true ]
 then
 	make_directory $ARRIBA_DIR
 	cd $ARRIBA_DIR
-	echo_run "$ARRIBA_READTHROUGH_BINARY -g $GENE_MODELS -i $ALIGNMENT_DIR/$STAR_SORTED_MKDUP_BAM -o ${SAMPLE}_${pid}_merged_read_through.bam"
-	echo_run "$ARRIBA_BINARY -c $ALIGNMENT_DIR/$STAR_CHIMERA_MKDUP_BAM -r ${SAMPLE}_${pid}_merged_read_through.bam -x $ALIGNMENT_DIR/$STAR_SORTED_MKDUP_BAM -a $GENOME_FA -k $ARRIBA_KNOWN_FUSIONS -g $GENE_MODELS -b $ARRIBA_BLACKLIST -o ${SAMPLE}_${pid}.fusions.txt -O ${SAMPLE}_${pid}.discarded_fusions.txt "
-	if [[ -f "${SAMPLE}_${pid}.fusions.txt" ]]
+
+	arribaCommand="$ARRIBA_BINARY -c '$ALIGNMENT_DIR/$STAR_CHIMERA_MKDUP_BAM' -x '$ALIGNMENT_DIR/$STAR_SORTED_MKDUP_BAM' -a '$GENOME_FA' -k '$ARRIBA_KNOWN_FUSIONS' -g '$GENE_MODELS' -b '$ARRIBA_BLACKLIST' -T -P -o '${SAMPLE}_$pid.fusions.txt' -O '${SAMPLE}_$pid.discarded_fusions.txt'"
```
Comment (suhrig) [inline resources/analysisTools/rnaseqworkflow/HIPO2_rnaseq_processing.sh]: How about adding `gzip -9 '${SAMPLE}_$pid.discarded_fusions.txt'` after the call to Arriba? It's a big plain-text file that most users do not look into anyway. I usually compress it to save some disk space.

File: resources/analysisTools/rnaseqworkflow/HIPO2_rnaseq_processing.sh [hunk 3]
```diff
@@ -262,11 +262,20 @@ if [ "$RUN_ARRIBA" == true ]
...
+    else
+        arribaCommand="$arribaCommand -s no"
+    fi
+
+    echo_run "$arribaCommand"
+
+	if [[ -f "${SAMPLE}_$pid.fusions.txt" ]]
 	then
-		echo_run "$ARRIBA_DRAW_FUSIONS --annotation=$GENE_MODELS --fusions=${SAMPLE}_${pid}.fusions.txt --output=${SAMPLE}_${pid}.fusions.pdf"
+		echo_run "$ARRIBA_DRAW_FUSIONS --annotation='$GENE_MODELS' --fusions='${SAMPLE}_$pid.fusions.txt' --output='${SAMPLE}_$pid.fusions.pdf'"
```
Comment (suhrig) [inline resources/analysisTools/rnaseqworkflow/HIPO2_rnaseq_processing.sh]: The new `draw_fusions.R` can do much more than the old version. But it needs some additional files for that. Here is the enhanced command:

```
echo_run "$ARRIBA_DRAW_FUSIONS --annotation='$GENE_MODELS' --proteinDomains='$ARRIBA_PROTEIN_DOMAINS' --cytobands='$ARRIBA_CYTOBANDS' --alignments='$ALIGNMENT_DIR/$STAR_SORTED_MKDUP_BAM' --fusions='${SAMPLE}_$pid.fusions.txt' --output='${SAMPLE}_$pid.fusions.pdf'"
```

`$ARRIBA_PROTEIN_DOMAINS` should be set to `/path/to/protein_domains_hg19_hs37d5_GRCh37_2019-07-05.gff3` and `$CYTOBANDS` to `/path/to/cytobands_hg19_hs37d5_GRCh37_2018-02-23.tsv`. The files can be found in the [Arriba release v1.2.0](https://github.com/suhrig/arriba/releases/download/v1.2.0/arriba_v1.2.0.tar.gz) in the `database` folder.

File: resources/analysisTools/rnaseqworkflow/HIPO2_rnaseq_processing.sh [hunk 4]
```diff
@@ -37,9 +37,9 @@ check_executable "$ARRIBA_READTHROUGH_BINARY"
 check_executable "$ARRIBA_DRAW_FUSIONS"
```
Comment (suhrig) [inline resources/analysisTools/rnaseqworkflow/HIPO2_rnaseq_processing.sh]: `check_executable "$ARRIBA_READTHROUGH_BINARY"` can be removed, as this executable no longer exists in Arriba version 1.2.0.

File: resources/analysisTools/rnaseqworkflow/HIPO2_rnaseq_processing.sh (hunk 2 above)
Comment (suhrig) [inline resources/analysisTools/rnaseqworkflow/HIPO2_rnaseq_processing.sh]: In addition, can you add the parameter `-I` to the Arriba call?. It causes the identifiers of supporting reads to be printed for each fusion prediction. Most people won't need this, but additional information does not hurt and it is a pain to have to rerun Arriba again, if someone ever does need this info.

Comment (vinjana) [reply resources/analysisTools/rnaseqworkflow/HIPO2_rnaseq_processing.sh]: Yes, I think this is a good suggestion. I'm going to add the call.

Comment (vinjana) [reply resources/analysisTools/rnaseqworkflow/HIPO2_rnaseq_processing.sh]: Yes, I think this is a good service-minded argument. I'm going to do this.

File: resources/analysisTools/rnaseqworkflow/HIPO2_rnaseq_processing.sh (hunk 1 above)
Comment (vinjana) [reply resources/analysisTools/rnaseqworkflow/HIPO2_rnaseq_processing.sh]: We spoke about it. Thanks for the reminder!

File: resources/analysisTools/rnaseqworkflow/HIPO2_rnaseq_processing.sh (hunk 3 above)
Comment (vinjana) [reply resources/analysisTools/rnaseqworkflow/HIPO2_rnaseq_processing.sh]: Many thanks for these suggestions!

File: resources/configurationFiles/analysisRNAseq.xml [hunk 5]
```diff
@@ -120,6 +120,8 @@
         <cvalue name='GENOME_KALLISTO_INDEX' value='${indexDirectory}/kallisto/kallisto-0.43.0_1KGRef_Gencode19_k31/kallisto-0.43.0_1KGRef_Gencode19_k31.noGenes.index' type='string'/>
         <cvalue name='ARRIBA_KNOWN_FUSIONS' value='${hg19BaseDirectory}/tools_data/arriba/known_fusions_CancerGeneCensus_gencode19_2017-01-16.tsv.gz' type='string'/>
         <cvalue name='ARRIBA_BLACKLIST' value='${hg19BaseDirectory}/tools_data/arriba/blacklist_hs37d5_gencode19_2017-01-09.tsv.gz' type='string'/>
```
Comment (suhrig) [inline resources/configurationFiles/analysisRNAseq.xml]: There are more recent versions of the blacklist and known fusions files that should be used with Arriba version 1.2.0. I did not notice this in the previous review, sorry. They weren't displayed in the diff ...

These are the correct versions, both of which already reside in ngs_share:

- `blacklist_hg19_hs37d5_GRCh37_2018-11-04.tsv.gz`
- `known_fusions_CancerGeneCensus_gencode19_2017-05-11.tsv.gz`

Everything else looks fine to me.

Comment (vinjana) [reply resources/configurationFiles/analysisRNAseq.xml]: Perfect that you noticed! I was not aware of this. I will add the new data as default. The files are, however, not part of this repo. So I will only mention them here in the XML and in the readme, such that users can install them themselves. I am very grateful for every comment, in particular if it is such a valuable one.

**Pull request merged**